Add zephyr control path

D-Triveni opened the pull request.

D-Triveni committed:
[nrf fromlist]: fixup! Add AP support flag

Upstream-PR: https://github.com/Wi-FiQuickTrack/Wi-FiQuickTrack-ControlAppC/pull/7

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

D-Triveni committed:
[nrf noup]: zephyr: qt: Add missing AP flag

Add missing AP flag in zephyr specific file

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

D-Triveni committed:
[nrf noup]: zephyr: qt: Add zephyr based control interface

Replaced Linux version of opening/closing control interface
due to incompatibility with zephyr for supplicant connectivity.

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

D-Triveni committed:
[nrf noup]: zephyr: qt: Remove unistd.h file

fixup! [nrf noup]: zephyr: qt: Add zephyr based control interface

When both unistd.h and net/socket.h are included, compiler
is listing below error for close() declaration.
error: static declaration of 'close' follows non-static declaration

To fix this, remove unistd.h as we need static declaration of
close() from net/socket.h

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

D-Triveni committed:
[nrf noup]: zephyr: qt: Fix conflicting types for inet_ntop

fixup! [nrf noup]: zephyr: qt: Add zephyr based control interface

Definition of inet_ntop here is conflicting with
the definition from net/socket.h file. To fix the
error, remove definition here and fetch it from
header file.

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

D-Triveni committed:
[nrf noup]: zephyr: qt: Replace sleep with k_sleep

fixup! [nrf noup]: zephyr: qt: Add zephyr based control interface

Need to include unistd.h for sleep() declaration, but
by inlcuding it other compilation errors are seen.
To fix this, use k_sleep().

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

D-Triveni committed:
[nrf noup]: zephyr: qt: Add wireless interface information

Set default wireless interface information.
Other API's read wireless interface information to
invoke control path.

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

krish2718 commented on zephyr/src/indigo_api_callback_dut.c:
```diff
@@ -1263,8 +1268,14 @@ static int get_mac_addr_handler(struct packet_wrapper *req, struct packet_wrappe
+
     if (atoi(role) == DUT_TYPE_STAUT) {
-        w = wpa_ctrl_open(get_wpas_ctrl_path());
+        w = wpa_ctrl_open(wpa_s->ctrl_iface->sock_pair[0]);
```
if you are removing close, why do we need to open everytime? Just use the global variable or add a wrapper to get the handle

D-Triveni replied:
@krish2718 As per the discussion, keeping the changes which are working fine.

krish2718 reviewed (changes requested):
17037b114164f42ec1c687b3a14017b1e4c9ecd1 the body needs more details, and please remove `add changes` and instead explain `what` brieflyy and more importantly `why`

D-Triveni committed:
[nrf fromlist]: Remove new line at EOF

Upstream-PR: https://github.com/Wi-FiQuickTrack/Wi-FiQuickTrack-ControlAppC/pull/10

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

D-Triveni committed:
[nrf noup]: zephyr: qt: Fix new line

Remove the newline character at the end of the file.
To meet the compliance requirements, there should be
no blank lines at the end of the file.

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>

D-Triveni committed:
[nrf noup] zephyr: qt: Update logger level handling

fixup! [nrf noup] zephyr: qt: Fix the logger issues

Previously, the logger level was set to LOG_LEVEL_DEFAULT,
but this configuration was not producing any visible logs.
So, the logger level has been changed to LOG_LEVEL to ensure
that logs are displayed based on user-defined settings.

Signed-off-by: Triveni Danda <triveni.danda@nordicsemi.no>
